src/main/java/net/neoforged/neoforge/common/util/AttributeUtil.java

```diff
@@ -0,0 +1,356 @@
+            amt += baseBonus;
+
+            boolean isMerged = NeoForgeMod.shouldMergeAttributeTooltips() && (!baseModif.children.isEmpty() || baseBonus != 0);
+            MutableComponent text = attr.value().toBaseComponent(amt, entityBase, isMerged, ctx.flag());
+            tooltip.accept(Component.literal(" ").append(text).withStyle(isMerged ? ChatFormatting.GOLD : ChatFormatting.DARK_GREEN));
+            if (ctx.flag().hasShiftDown() && isMerged) {
+                // Display the raw base value, and then all children modifiers.
+                text = attr.value().toBaseComponent(rawBase, entityBase, false, ctx.flag());
+                tooltip.accept(AttributeUtil.list().append(text.withStyle(ChatFormatting.DARK_GREEN)));
+                for (AttributeModifier modifier : baseModif.children) {
+                    tooltip.accept(AttributeUtil.list().append(attr.value().toComponent(modifier, ctx.flag())));
+                }
+                if (baseBonus > 0) {
+                    attr.value().addBonusTooltips(stack, tooltip, ctx.flag());
+                }
+            }
+        }
+
+        for (Holder<Attribute> attr : modifierMap.keySet()) {
+            // Skip attributes who have already been processed during the base modifier stage
+            if (NeoForgeMod.shouldMergeAttributeTooltips() && baseModifs.containsKey(attr)) {
+                continue;
+            }
+
+            Collection<AttributeModifier> modifs = modifierMap.get(attr);
+            // Initiate merged-tooltip logic if we have more than one modifier for a given attribute.
+            if (NeoForgeMod.shouldMergeAttributeTooltips() && modifs.size() > 1) {
+                double[] sums = new double[3];
+                boolean[] merged = new boolean[3];
+                Map<Operation, List<AttributeModifier>> shiftExpands = new HashMap<>();
+
+                for (AttributeModifier modifier : modifs) {
+                    if (modifier.amount() == 0) continue;
+                    if (sums[modifier.operation().ordinal()] != 0) merged[modifier.operation().ordinal()] = true;
+                    sums[modifier.operation().ordinal()] += modifier.amount();
+                    shiftExpands.computeIfAbsent(modifier.operation(), k -> new LinkedList<>()).add(modifier);
+                }
+
+                for (Operation op : Operation.values()) {
+                    int i = op.ordinal();
+
+                    // If the merged value comes out to 0, just ignore the whole stack
+                    if (sums[i] == 0) {
+                        continue;
+                    }
+
+                    // Handle merged modifier stacks by creating a "fake" merged modifier with the underlying value.
+                    if (merged[i]) {
+                        TextColor color = attr.value().getMergedStyle(sums[i] > 0);
+                        var fakeModif = new AttributeModifier(AttributeUtil.FAKE_MERGED_ID, sums[i], op);
```

Comment:
We are already in this class, we don't need to explicitly reference `AttributeUtil`